File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/social/RVSocialListAdapter.java
package com.mentalmachines.droidcon_boston.views.social;

import android.content.Context;
import android.support.v7.widget.RecyclerView.Adapter;
import android.support.v7.widget.RecyclerView.ViewHolder;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import com.mentalmachines.droidcon_boston.R;
import com.mentalmachines.droidcon_boston.modal.SocialModal;
import java.util.ArrayList;

class RVSocialListAdapter extends Adapter<RVSocialListAdapter.ListViewHolder> {

  public class ListViewHolder extends ViewHolder {

    ImageView imageView;

    TextView txtView;

    public ListViewHolder(final View itemView) {
      super(itemView);
      imageView = itemView.findViewById(R.id.social_item_img);
      txtView = itemView.findViewById(R.id.social_item_tv);
    }
  }

  ArrayList<SocialModal> socialList;


  public RVSocialListAdapter(final ArrayList<SocialModal> socialList) {
    this.socialList = socialList;
  }

  @Override
  public int getItemCount() {
    return socialList.size();
  }

  @Override
  public void onBindViewHolder(final ListViewHolder holder, final int position) {
    SocialModal socialModal = socialList.get(position);

    holder.txtView.setText(socialModal.getName());

    holder.imageView.setImageResource(socialModal.getImage_resid());

  }

  @Override
  public ListViewHolder onCreateViewHolder(final ViewGroup parent, final int viewType) {
    Context context = parent.getContext();
    View view = LayoutInflater.from(context).inflate(R.layout.social_list_item, parent, false);
    return new ListViewHolder(view);
  }
}

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/social/RVSocialListAdapter.kt
package com.mentalmachines.droidcon_boston.views.social

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.mentalmachines.droidcon_boston.R
import com.mentalmachines.droidcon_boston.modal.SocialModal
import java.util.*

internal class RVSocialListAdapter(private var socialList: ArrayList<SocialModal>) :
    Adapter<RVSocialListAdapter.ListViewHolder>() {

    inner class ListViewHolder(itemView: View) : ViewHolder(itemView) {

        internal var imageView: ImageView = itemView.findViewById(R.id.social_item_img)

        internal var txtView: TextView

        init {
            txtView = itemView.findViewById(R.id.social_item_tv)
        }
    }

    override fun getItemCount(): Int {
        return socialList.size
    }

    override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
        val socialModal = socialList[position]

        holder.txtView.text = socialModal.name

        holder.imageView.setImageResource(socialModal.image_resid)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewHolder {
        val context = parent.context
        val view = LayoutInflater.from(context).inflate(R.layout.social_list_item, parent, false)
        return ListViewHolder(view)
    }
}
